File: cppStuff/algorithms/RepeatedString.cpp
/*
Lilah has a string, s, of lowercase English letters that she repeated infinitely many times.

Given an integer, n, find and print the number of letter a's in the first  n letters of Lilah's infinite string.
*/

#include <iostream>
#include <string>

long long solve(std::string s,long long n){
    
    long long count = 0;
    for(char c : s)
        if(c == 'a')
            count++;
    
    long long j = n/(s.length());
    count *= j;
    
    long long k = n % s.length();
    for(int i = 0; i < k; i++)
        if(s[i] == 'a')
            count++;
    
    return count;
}

int main(){
    std::string s;
    std::cin >> s;
    long long n;
    std::cin >> n;
    long long results = solve(s, n);
    std::cout << results << '\n';
    return 0;
}

File: jStuff/RepeatedString.java
/*
Lilah has a string, s, of lowercase English letters that she repeated infinitely many times.

Given an integer, n, find and print the number of letter a's in the first  n letters of Lilah's infinite string.
*/

import java.util.Scanner;

public class Solution {

    private static long solve(String s, long n){
        long count = 0;
        for(char c : s.toCharArray())
            if(c == 'a')
                count++;
        
        long j = n / (s.length());
        count *= j;
        
        long k = n % s.length();
        for(int i = 0; i < k; i++)
            if(s.charAt(i) == 'a') 
                count++;
        
        return count;
    }
    
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String s = in.next();
        long n = in.nextLong();
        long results = solve(s, n);
        System.out.println(results);
    }
}
